protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1515,6 +1515,34 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 						copy(operationObject.Produces, opts.Produces)
 					}
 
+					if params := opts.Parameters; params != nil && len(params.Headers) > 0 {
+						for _, header := range params.Headers {
+							param := openapiParameterObject{
+								In:          "header",
+								Name:        header.Name,
+								Description: header.Description,
+								Required:    header.Required,
+								Format:      header.Format,
+							}
+
+							switch header.Type {
+							case openapi_options.HeaderParameter_STRING:
+								param.Type = "string"
+							case openapi_options.HeaderParameter_NUMBER:
+								param.Type = "number"
+							case openapi_options.HeaderParameter_INTEGER:
+								param.Type = "integer"
+							case openapi_options.HeaderParameter_BOOLEAN:
+								param.Type = "boolean"
+							default:
+								// TODO(krak3n): Should we error or assume string?
+								param.Type = "string"
```

Comment:
This should be an error per https://swagger.io/specification/v2/#parameterType which says this is required. I think we should make the user pick a type.

Reply:
Added an error, I wanted to test this but looks like we panic in `applyTemplate` if `renderServices` fails which makes it quite tricky to test the returned error since I'd need to defer & recover etc. Is it ok to not have a test case for this?

Reply:
Yeah this is fine 😁